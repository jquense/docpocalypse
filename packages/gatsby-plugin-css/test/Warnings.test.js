const { webpack } = require('./helpers');

const plugin = () => (css, result) => {
  css.walkDecls((node) => {
    node.warn(result, '<Message>');
  });
};

describe('Warnings', () => {
  test('Plugins', () => {
    const config = {
      loader: {
        test: /\.css$/,
        options: {
          plugins: [plugin()],
        },
      },
    };

    return webpack('css/index.js', config).then((stats) => {
      const warning = stats.compilation.warnings[0];

      const message = warning.message.split('\n').slice(1).join('\n');

      expect(message).toMatchSnapshot();
    });
  });
});
