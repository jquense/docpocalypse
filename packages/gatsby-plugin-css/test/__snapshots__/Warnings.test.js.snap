// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Warnings Plugins 1`] = `
"Warning

(1:5) <Message>"
`;
